apps/flame-defi/app/pool/types.ts

```diff
@@ -67,26 +67,56 @@ export type Positions = {
   [key: number]: Position;
 };
 
+export type PoolPositionResponse = {
+  nonce: bigint;
+  operator: string;
+  tokenAddress0: Address;
+  tokenAddress1: Address;
+  fee: number;
+  tickLower: number;
+  tickUpper: number;
+  liquidity: bigint;
+  feeGrowthInside0LastX128: bigint;
+  feeGrowthInside1LastX128: bigint;
+  tokensOwed0: bigint;
+  tokensOwed1: bigint;
+  tokenId?: string;
+};
+
+export interface PoolPosition extends PoolPositionResponse {
+  feePercent: FeeTier;
+  symbolOne: string;
```

Comment:
Why do you need symbolOne/Two and tokenOne/Two?

Reply:
This what we can handle the null checks for symbols in context and we don't have to have that problem lower down. I would rather have context handle as much data formatting as possible instead of doing it at the component level.

Reply:
I'm not sure I follow. It looks like it always describes the same data. Not too different from not needing `poolTokens` and `poolTokenOne`/`Two`

```
    return {
          symbolOne: tokenOne?.coinDenom ?? "",
          symbolTwo: tokenTwo?.coinDenom ?? "",
          feePercent,
          inRange: !isClosed,
          positionStatus: isClosed ? "Closed" : "In range",
          poolAddress,
          tokenOne,
          tokenTwo,
          ...position,
        };
```

`symbolOne`/`Two` are always going to be `""` when `tokenOne`/`Two` is `null`. This just duplicates information and makes the code harder to read I think.

What do you mean by "instead of doing it at the component level"?